src/dataServices/dynamoDbBundleServiceHelper.ts

```diff
@@ -253,4 +256,211 @@ export default class DynamoDbBundleServiceHelper {
+                }
+                case 'delete': {
+                    if (vid === 0) {
+                        break;
+                    }
+                    deleteRequests.push({
+                        Statement: `
+                            UPDATE "${RESOURCE_TABLE}"
+                            SET "${DOCUMENT_STATUS_FIELD}" = '${DOCUMENT_STATUS.DELETED}'
+                            WHERE "id" = '${buildHashKey(id, tenantId)}' AND "vid" = ${vid}
+                        `,
+                        id,
+                    });
+
+                    batchReadWriteResponses.push({
+                        id,
+                        vid: vid.toString(),
+                        operation: request.operation,
+                        lastModified: new Date().toISOString(),
+                        resource: {},
+                        resourceType: request.resourceType,
+                    });
+                    break;
+                }
+                case 'read': {
+                    if (vid === 0) {
+                        break;
+                    }
+                    batchReadWriteResponses.push({
+                        id,
+                        vid: vid.toString(),
+                        operation: request.operation,
+                        lastModified: '',
+                        resource: item?.resource,
+                        resourceType: request.resourceType,
+                    });
+                    break;
+                }
+                default:
+                    break;
+            }
+        }
+        // we cannot do deleteRequests nor updateRequests in a batchwriteitem call, since we use the update api instead of delete
+        // hence, we will separate these requests and use batchexecuteStatement to update items in a batch (and
+        // we know there are no conflicts since there will only be updates running)
+        return {
+            editRequests: [...deleteRequests, ...createRequests, ...updateRequests],
+            batchReadWriteResponses,
+        };
+    }
+
+    static async processBatchEditRequests(
+        editRequests: any[],
+        batchReadWriteResponses: BatchReadWriteResponse[],
+        dynamoDb: DynamoDB,
+    ) {
+        const updatedResponses = batchReadWriteResponses;
+        for (let i = 0; i < editRequests.length; i += MAX_BATCH_WRITE_ITEMS) {
+            const upperLimit = Math.min(i + MAX_BATCH_WRITE_ITEMS, editRequests.length);
+            const batch = editRequests.slice(i, upperLimit);
+            const statements = batch.map((x) => {
+                return { Statement: x.Statement };
+            });
+            // eslint-disable-next-line no-await-in-loop
+            const batchExecuteResponse = await dynamoDb
+                .batchExecuteStatement({
+                    Statements: [...statements],
+                })
+                .promise();
+            batchExecuteResponse?.Responses?.forEach((response, index) => {
```

Comment:
When would the Response be empty? Is it when there's no write requests?

Reply:
I think this was a holdover from trying to solve the issue with async/await in a try/catch, the `batchExecuteResponse` will never be empty in this scenario.

Reply:
Emmm so what would happen if the question marks are removed?

Reply:
Oh wait nvm this snippet is out of date. The question mark is gone in the PR